### PR TITLE
feat: pom xml latest version update and xml format change

### PR DESCRIPTION
**Pull request by Joel-hanson**
- updated pom.xml to latest version
- updated xml format to be more consistent
- updated Dockerfile to use latest version of kafka
- New Dockerfile with updated versions of the base image
and the packages installed in the image.
- java distribution and pom package version updates
- Update java distribution to semeru from temurin
- Update com.ibm.mq.allclient to 9.3.2.1 to fix
  SNYK-JAVA-ORGJSON-5488379
- Update mockito-core to 4.11.0 to support java 8
- Update mq and kafka version support in README.md
- Update version from 1.3.3 to 1.3.4

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -14,254 +14,266 @@
...
+    </licenses>
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+    </properties>
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.kafka</groupId>
+            <artifactId>connect-api</artifactId>
+            <version>3.4.0</version>
```
Comment (dalelane) [inline pom.xml:49]: Would a jar built using a newer version of Kafka still work in a Connect runtime using an older version of Kafka? 

(Apologies - I should know the answer to this!)

If yes - great. 
If not - we should at least update the README to specify the minimum version of Connect required to use the connector if we make this change.

**Comment by david-VigneshN** [comment]
Just curious on the feature introduced in Apache Kafka version 3.3.1 for KIP-618: Exactly-Once support for source connectors.
Did we have to do any changes in our connector code to adapt this feature?

**Comment by dalelane** [comment]
@david-VigneshN Exactly-once support is something that I've been looking at - as you point out, the newer versions of Kafka Connect do make this possible. This isn't completely "free" as some code changes will be needed in order to offer assurances for exactly-once support, but this is something I'm interested in. 

I'd be interested to know how important/useful that would be for your use case(s) if you'd be willing to share any details.

**Comment by david-VigneshN** [comment]
@dalelane I was curious to know what changes it would take and good to have it added. It is not a mandatory for our current use cases. 
We are using this source connectors to transmit messages from our on-premise MQ to Kafka/Azure event hub.

File: pom.xml (hunk 1 above)
Comment (Joel-hanson) [reply pom.xml:49]: Yes, it does work with older version of kafka connect runtime. I was able to test it with kafka-2.6.2 and higher also having mq 9 or higher. I have updated the readme with the same.

Comment (Joel-hanson) [reply pom.xml:49]: @dalelane ⬆️

**Review by dalelane — DISMISSED** [review]

**Review by jhughes24816 — APPROVED** [review]
LGTM thanks Joel

**Pull request merged**